src/vivarium_census_prl_synth_pop/utilities.py

```diff
@@ -500,3 +500,47 @@ def write_to_disk(data: pd.DataFrame, path: Path):
             f"Supported extensions are {metadata.SUPPORTED_EXTENSIONS}. "
             f"{path.suffix[1:]} was provided."
         )
+
+
+def copy_from_household_member(
+    pop: pd.DataFrame, randomness_stream: RandomnessStream
+) -> pd.DataFrame:
+    # Creates copy_age, copy_date_of_birth, and copy_ssn from household members
+    # Note: copy value can be original value but copies from another household member
+
+    copy_cols = metadata.COPY_HOUSEHOLD_MEMBER_COLS
+    for col in [column for column in copy_cols.keys() if column in pop.columns]:
+        pop[copy_cols[col]] = np.nan
+        if col == "has_ssn":
+            # Subset to rows that have SSN so we do not try and map nans.
+            households = (
+                pop[pop[col]].groupby("household_id").apply(lambda df_g: list(df_g.index))
+            )
+        else:
+            # This makes the assumption age and date of birth will not have nans at this poitn
+            households = pop.groupby("household_id").apply(lambda df_g: list(df_g.index))
+        # Drop GQ and households with single member
+        households = households[households.apply(len) > 1]
+        households = households.loc[
+            households.index.difference(data_values.GQ_HOUSING_TYPE_MAP.keys())
+        ]
+        simulants_and_household_members = pop["household_id"].map(households).dropna()
```

Comment:
Do you need the .dropna()? Wouldn't the closest to an NaN be an empty list from households?

Reply:
I believe you are right. We also drop lists that are less than length 2 so I think this is unnecessary but the idea was to prevent tryign to map nans.